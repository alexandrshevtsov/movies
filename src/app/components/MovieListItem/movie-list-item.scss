@import "../../global.scss";

.movieListItem {
  display: flex;
  background-color: $content-block-background-color;
  border: 1px solid $content-block-border-color;
  border-radius: $content-block-border-radius;
  margin-bottom: $content-block-spacing;
  padding: $content-block-padding-2;

  .imageContainer {
    display: inline-block;

    .image {
      height: 346px;
      width: 250px;
    }
  }

  .infoContainer {
    display: inline-block;
    padding-left: $content-block-padding-2;
    position: relative;

    .detailsLinkContainer {
      width: 98%;
      text-align: center;
      position: absolute;
      bottom: 0;
    }
  }
}

@media screen and (max-width: 600px) {
  .movieListItem {
    display: block;

    .imageContainer {
      display: block;

      .image {
        height: 346px;
        width: 250px;
      }
    }

    .infoContainer {
      p, br, .detailsLinkContainer {
        display: none;
      }
    }
  }
}
