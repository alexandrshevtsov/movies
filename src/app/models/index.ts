export * from './movie-list-page-model';
export * from './movie-details-model';
