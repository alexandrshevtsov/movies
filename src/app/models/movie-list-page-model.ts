import { MovieListItemModel } from './movie-list-item-model';

export interface MovieListPageModel {
  page: number;
  total_results: number;
  total_pages: number;
  results: MovieListItemModel[];
}
