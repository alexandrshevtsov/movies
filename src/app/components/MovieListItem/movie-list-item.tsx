import * as React from 'react';
import { MovieListItemModel } from '../../models/movie-list-item-model';
import LazyLoad from 'react-lazyload';
import { Link } from 'react-router-dom';
import styles from './movie-list-item.scss';

interface IMovieListItemProps {
  model: MovieListItemModel;
}

export const MovieListItem = (props: IMovieListItemProps) => {
  const detailsUrl = `/details/${props.model.id}`;
  return (
    <div className={styles.movieListItem}>
      <div className={styles.imageContainer}>
        <LazyLoad>
          <Link to={detailsUrl} target="_blank">
            <img
              className={styles.image}
              alt=""
              src={`https://image.tmdb.org/t/p/w500/${props.model.poster_path}`}
            />
          </Link>
        </LazyLoad>
      </div>
      <div className={styles.infoContainer}>
        <h1>{props.model.title}</h1>
        <p>{props.model.overview}</p>
        <br />
        <br />
        <div className={styles.detailsLinkContainer}>
          <hr />
          <Link className={styles.detailsLink} to={detailsUrl} target="_blank">
            MORE INFO
          </Link>
        </div>
      </div>
    </div>
  );
};
