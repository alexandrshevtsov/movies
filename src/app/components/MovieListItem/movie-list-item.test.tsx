import * as React from 'react';
import { mount } from 'enzyme';
import { MovieListItem } from './movie-list-item';
import { MovieListItemModel } from '../../models/movie-list-item-model';
import { BrowserRouter as Router } from 'react-router-dom';

describe('MovieListItem', () => {
  it('rendering', () => {
    var model: MovieListItemModel = {
      id: 123,
      title: 'My Movie',
      overview: 'My Movie overview',
      poster_path: 'image.jpg'
    } as MovieListItemModel;
    const component = mount(
      <Router>
        <MovieListItem model={model} />
      </Router>
    );
    const titleComponent = component.find('h1');
    expect(titleComponent.contains('My Movie')).toBe(true);
    //const overviewComponent = component.find('');
    //expect(overviewComponent.contains('My Movie overview')).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
