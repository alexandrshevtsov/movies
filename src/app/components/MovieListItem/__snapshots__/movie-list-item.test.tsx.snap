// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieListItem rendering 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Component
      model={
        Object {
          "id": 123,
          "overview": "My Movie overview",
          "poster_path": "image.jpg",
          "title": "My Movie",
        }
      }
    >
      <div>
        <div>
          <LazyLoad
            offset={0}
            once={false}
            overflow={false}
            resize={false}
            scroll={true}
            unmountIfInvisible={false}
          >
            <div
              className="lazyload-placeholder"
              style={
                Object {
                  "height": undefined,
                }
              }
            />
          </LazyLoad>
        </div>
        <div>
          <h1>
            My Movie
          </h1>
          <p>
            My Movie overview
          </p>
          <br />
          <br />
          <div>
            <hr />
            <Link
              target="_blank"
              to="/details/123"
            >
              <a
                href="/details/123"
                onClick={[Function]}
                target="_blank"
              >
                MORE INFO
              </a>
            </Link>
          </div>
        </div>
      </div>
    </Component>
  </Router>
</BrowserRouter>
`;
